### PR TITLE
Add typing to dimension separator arguments

### PR DESCRIPTION
**Pull request by dstansby**
Not very exciting, but carrys on the typing train. This PR adds typing to wherever a dimension separator argument is used in functions/methods.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Review by d-v-b — APPROVED** [review]

**Comment by d-v-b** [comment]
thank you! I love the typing train

**Comment by dstansby** [comment]
Is there anything more I need to do here before this can be merged?

File: zarr/storage.py
Comment (d-v-b) [inline zarr/storage.py:2086]: this is not a blocker, but we should look into making `NoLock` an instance of `Lock`

**Comment by d-v-b** [comment]
it looks good to me! unless anyone objects, I'm going to hit the merge button later today

**Pull request merged**
